src/modules/Game.class.js

```diff
@@ -50,19 +91,121 @@ class Game {
    * `win` - the game is won;
    * `lose` - the game is lost
    */
-  getStatus() {}
+  getStatus() {
+    return this.status;
+  }
 
   /**
    * Starts the game.
    */
-  start() {}
+  start() {
+    this.status = 'playing';
+    this.addRandomTile();
+    this.addRandomTile();
+  }
 
   /**
    * Resets the game.
    */
-  restart() {}
+  restart() {
+    this.status = 'idle';
+    this.board = this.boardDeepCopy(this.initialState);
+    this.score = 0;
+  }
 
   // Add your own methods here
+
+  addRandomTile() {
+    const emptyCells = this.board.reduce((acc, row, rowIndex) => {
+      row.forEach((cell, colIndex) => {
+        if (cell === 0) {
+          acc.push({ row: rowIndex, col: colIndex });
+        }
+      });
+
+      return acc;
+    }, []);
+
+    if (emptyCells.length > 0) {
+      const { row, col } =
+        emptyCells[Math.floor(Math.random() * emptyCells.length)];
+
+      this.board[row][col] = Math.random() < 0.9 ? 2 : 4;
+    }
+
+    if (this.isGameOver()) {
+      this.status = 'lose';
+    }
```

Comment:
The `addRandomTile` method checks for game over after adding a tile. This is correct, but ensure that the game status is also updated appropriately elsewhere in the code, especially after moves, to reflect win conditions.